benchmark/benchmarks_entrypoint.py

```diff
@@ -18,61 +38,239 @@ class ImportModuleException(Exception):
+        if self.use_database:
+            self.conn.autocommit = True
         self.logger = logger
         self.repository = repository
         self.branch = branch
         self.commit_id = commit_id
         self.commit_msg = commit_msg
+        self.collect_csv_data = collect_csv_data
+        
+        # For CSV export - store all data in pandas DataFrames (only if CSV collection is enabled)
+        if self.collect_csv_data:
+            # Initialize empty DataFrames with proper schemas
+            self.benchmarks_df = pd.DataFrame(columns=[
+                'benchmark_id', 'repository', 'branch', 'commit_id', 'commit_message', 
+                'metadata', 'created_at'
+            ])
+            self.device_measurements_df = pd.DataFrame(columns=[
+                'benchmark_id', 'cpu_util', 'mem_megabytes', 'gpu_util', 
+                'gpu_mem_megabytes', 'time'
+            ])
+            self.model_measurements_df = pd.DataFrame(columns=[
+                'benchmark_id', 'time', 'model_load_time', 'first_eager_forward_pass_time_secs',
+                'second_eager_forward_pass_time_secs', 'first_eager_generate_time_secs',
+                'second_eager_generate_time_secs', 'time_to_first_token_secs',
+                'time_to_second_token_secs', 'time_to_third_token_secs',
+                'time_to_next_token_mean_secs', 'first_compile_generate_time_secs',
+                'second_compile_generate_time_secs', 'third_compile_generate_time_secs',
+                'fourth_compile_generate_time_secs'
+            ])
+        else:
+            self.benchmarks_df = None
+            self.device_measurements_df = None
+            self.model_measurements_df = None
 
-    def initialise_benchmark(self, metadata: dict[str, str]) -> int:
+    def initialise_benchmark(self, metadata: dict[str, str]) -> str:
         """
-        Creates a new benchmark, returns the benchmark id
+        Creates a new benchmark, returns the benchmark id (UUID)
         """
-        # gpu_name: str, model_id: str
-        with self.conn.cursor() as cur:
-            cur.execute(
-                "INSERT INTO benchmarks (repository, branch, commit_id, commit_message, metadata) VALUES (%s, %s, %s, %s, %s) RETURNING benchmark_id",
-                (self.repository, self.branch, self.commit_id, self.commit_msg, metadata),
-            )
-            benchmark_id = cur.fetchone()[0]
-            logger.debug(f"initialised benchmark #{benchmark_id}")
-            return benchmark_id
-
-    def collect_device_measurements(self, benchmark_id: int, cpu_util, mem_megabytes, gpu_util, gpu_mem_megabytes):
+        # Generate a unique UUID for this benchmark
+        benchmark_id = str(uuid.uuid4())
+        
+        if self.use_database:
+            with self.conn.cursor() as cur:
+                cur.execute(
+                    "INSERT INTO benchmarks (benchmark_id, repository, branch, commit_id, commit_message, metadata) VALUES (%s, %s, %s, %s, %s, %s)",
```

Comment:
FYI this won't work as the db schema expects an integer for the benchmark_id. We can change this if need be, there is a UUID type in postgres.